Add sample Android app

msridhar opened the pull request:
Fixes #59

lazaroclapp commented on sample-app/src/main/java/com/uber/myapplication/MainActivity.java:
```diff
@@ -0,0 +1,31 @@
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+    // Object x = null;
+    // x.hashCode();
```
Why these commented lines?

msridhar replied:
To test that NullAway was actually working.  If you uncomment them you get a compile error.

msridhar replied:
I'll add a comment

kageiit commented on gradle/dependencies.gradle:
```diff
@@ -35,6 +36,22 @@ def build = [
     jsr305Annotations       : "com.google.code.findbugs:jsr305:3.0.2",
+
+    // android stuff
+    buildToolsVersion: "25.0.2",
```
lets use 27

kageiit commented on gradle/dependencies.gradle:
```diff
@@ -35,6 +36,22 @@ def build = [
+
+    // android stuff
+    buildToolsVersion: "25.0.2",
+    compileSdkVersion: 25,
```
lets use 27

kageiit commented on gradle/dependencies.gradle:
```diff
@@ -35,6 +36,22 @@ def build = [
+    compileSdkVersion: 25,
+    ci: "true" == System.getenv("CI"),
+    minSdkVersion: 16,
+    targetSdkVersion: 25,
```
lets use 27

kageiit commented on sample-app/build.gradle:
```diff
@@ -0,0 +1,66 @@
+        targetCompatibility JavaVersion.VERSION_1_7
+    }
+
+    // Setup a simple lint config for an android app/library.
```
dont need this for sample

kageiit commented on sample-app/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">My Application</string>
```
Make the string `Nullaway Sample`

kageiit commented on sample-app/gradle.properties:
```diff
@@ -0,0 +1,19 @@
+# limitations under the License.
+#
+
+POM_NAME=YOUR PROJECTS NAME
```
we can delete this file

hzsweers commented on gradle/dependencies.gradle:
```diff
@@ -35,6 +36,22 @@ def build = [
+    targetSdkVersion: 27,
+
+    gradlePlugins: [
+        android: "com.android.tools.build:gradle:2.3.1",
```
You should use 3.0.1, which is the current stable release. AutoDispose is on an older one but only because I haven't updated it yet :)

hzsweers commented on sample-app/build.gradle:
```diff
@@ -0,0 +1,64 @@
+buildscript {
+    dependencies {
+        classpath deps.build.gradlePlugins.android
+        classpath "net.ltgt.gradle:gradle-errorprone-plugin:0.0.11"
```
0.0.13

hzsweers commented on sample-app/build.gradle:
```diff
@@ -0,0 +1,64 @@
+
+    annotationProcessor "com.uber.nullaway:nullaway:0.2.2"
+
+    errorprone "com.google.errorprone:error_prone_core:2.1.1"
```
Shouldn't this be 2.1.3 now?

hzsweers commented on sample-app/src/test/java/com/uber/myapplication/MainActivityTest.java:
```diff
@@ -0,0 +1,28 @@
+ */
+package com.uber.myapplication;
+
+import static junit.framework.Assert.assertEquals;
```
You can remove this file

msridhar commented:
@kageiit @hzsweers think I addressed all your comments.  Note that I bumped to AGP 3.0.1 and NullAway still works fine.  If someone could help me fix the Travis build that would be appreciated

msridhar commented:
Ok finally fixed the Travis build

hzsweers commented:
LGTM. You can look at autodispose's handling of license stuff if it gives you trouble in the future I think
